lib/pure/os.nim

```diff
@@ -1476,6 +1476,24 @@ when defined(macosx):
   proc getExecPath2(c: cstring, size: var cuint32): bool {.
     importc: "_NSGetExecutablePath", header: "<mach-o/dyld.h>".}
 
+when defined(haiku):
+  const
+    PATH_MAX = 1024
+    B_FIND_PATH_IMAGE_PATH = 1000
+
+  proc find_path(codePointer: pointer, baseDirectory: cint, subPath: cstring,
+                 pathBuffer: cstring, bufferSize: csize): int32
+                {.importc, header: "<FindDirectory.h>"}
+
+  proc getApplHaiku(): string =
+    let buf = cstring createU(char, PATH_MAX)
```

Comment:
That's bad, instead use something like:

```nim

result = newString(PATH_MAX)
if find_path(nil, B_FIND_PATH_IMAGE_PATH, nil, cstring(result), PATH_MAX) == 0:
  let realLen = len(cstring(result))
  setLen(result, realLen)
else:
  result = ""
```

Reply:
Oh, never know you could use string like that, thanks!